docs/concepts/api-extension/custom-resources.md

```diff
@@ -72,6 +223,318 @@ implementations for your custom resources by writing and deploying your own stan
+
+   </td>
+   <td>You need to handle multiple versions of your API.<ul>
+
+<li>E.g. when developing an extension to share with the world.</li></ul>
+
+   </td>
+  </tr>
+</table>
+
+
+
+<table>
+  <tr>
+   <td colspan="4" ><strong>Flexibility Comparison</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Feature
+   </td>
+   <td>What it does
+   </td>
+   <td>CRDs
+   </td>
+   <td>Aggregated API
+   </td>
+  </tr>
+  <tr>
+   <td>Validation
+   </td>
+   <td rowspan="2" >These help users prevent errors and allow you to evolve your API independently of your clients.  
+<p>
+These features are most useful when there are many clients, and they can't all update at the same time.
+   </td>
+   <td>Alpha feature of CRDs in v1.8.  Checks limited to what is supported by OpenAPI v3.0.
+   </td>
+   <td>Yes, arbitrary validation checks.
+   </td>
+  </tr>
+  <tr>
+   <td>Defaulting
+   </td>
+   <td>No.  But can get same effect with an Initializer. (requires programming).
+   </td>
+   <td>Yes
+   </td>
+  </tr>
+  <tr>
+   <td>Multi-versioning
+   </td>
+   <td>Allows serving the same object through two API versions.  Can help ease API changes like renaming fields.
+<p>
+Less important if you control your client versions.
+   </td>
+   <td>No.
+   </td>
+   <td>Yes
+   </td>
+  </tr>
+  <tr>
+   <td>Custom Storage
+   </td>
+   <td>If you need storage with a different performance mode (e.g. time-series database instead of key-value store) or isolation for security (e.g. encryption secrets or different trust level for admins) or fault isolation.
+   </td>
+   <td>No.
+   </td>
+   <td>Yes.
+   </td>
+  </tr>
+  <tr>
+   <td>Custom Business Logic
+   </td>
+   <td>Do arbitrary checks or actions when creating, reading, updating or deleting an object.
+   </td>
+   <td>No.  But can get some of the same effects with Initializers or Finalizers. (requires programming).
+   </td>
+   <td>Yes.
+   </td>
+  </tr>
+  <tr>
+   <td>Subresources
+   </td>
+   <td>Add extra operations other than CRUD, such as "scale" or "exec".
+<p>
+Allows systems like like HorizontalPodAutoscaler and PodDisruptionBudget interact with your new resource. 
+<p>
+Allows splitting responsibility for setting spec vs status.
+<p>
+Allows incrementing object Generation on custom resource data mutation (requires Spec/Status split).
+   </td>
+   <td>No but planned.
+   </td>
+   <td>Yes.  Any Subresource.
+   </td>
+  </tr>
+  <tr>
+   <td>strategic-merge-patch
+   </td>
+   <td>The new endpoints support PATCH with <code>Content-Type: application/strategic-merge-patch+json</code>.
+<p>
+Useful for updating objects that may be modified both locally, and by the server.   See: <a href="https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-strategic-merge-patch-to-update-a-deployment">https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-strategic-merge-patch-to-update-a-deployment</a>  
+   </td>
+   <td>No
```

Comment:
This is feasible for CRD via OpenAPI schema, and is on the long-term roadmap, but there are no immediate plans as far as I know.